### PR TITLE
[#72]  Describe Message Handler Interceptors for Event Processors

### PR DESCRIPTION
**Pull request by smcvb**
This PR introduces a `Handler Interceptors` section to the `event-processing.md` file, describing what it does and how you can create and register one to your `EventProcessor`s.
I have additionally made a small adjustment to the handler interceptor description of the command and query section, fine tuning the usage of the `UnitOfWork` in there.

Lastly I have linked to the `event interceptors` section in the `metrics-and-monitoring.md` section, as that part was missing there whilst the query and command side links were represented.

This PR resolves issue #72

### CODE REVIEW

File: part-iii-infrastructure-components/query-processing.md [hunk 1]
```diff
@@ -48,7 +48,9 @@ The BeanValidationInterceptor also implements `MessageHandlerInterceptor`, allow
 
 Message Handler Interceptors can take action both before and after query processing. Interceptors can even block query processing altogether, for example for security reasons.
 
-Interceptors must implement the `MessageHandlerInterceptor` interface. This interface declares one method, `handle`, that takes three parameters: the query message, the current `UnitOfWork` and an `InterceptorChain`. The `InterceptorChain` is used to continue the dispatching process.
+Interceptors must implement the `MessageHandlerInterceptor` interface. 
+This interface declares one method, `handle`, that takes three parameters: the query message, the current `UnitOfWork` and an `InterceptorChain`. 
+The `InterceptorChain` is used to continue the dispatching process, where as the `UnitOfWork` gives you (1) the message being handled and (2) provides the possibility to tie in logic prior, during or after (query) message handling (see [UnitOfWork](../part-i-getting-started#unit-of-work) for more information about the phases).
```
Comment (abuijze) [inline part-iii-infrastructure-components/query-processing.md]: where as --> whereas

File: part-iv-advanced-tuning/metrics-and-monitoring.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ public class MonitoringConfiguration {
 ### Interceptor Logging
 
 Another good approach to track the flow of messages throughout an Axon application is by setting up the right interceptors in your application.  
-As you might now there are two flavors of interceptors, the Dispatch and Handler interceptors (like discussed [here](../part-iii-infrastructure-components/command-dispatching.md#command-interceptors) for commands and [here](../part-iii-infrastructure-components/query-processing.md#handler-interceptors) for queries), which intercept a message prior to publishing (Dispatch Interceptor) or whilst it is being handled (Handler Interceptor).
+As you might now there are two flavors of interceptors, the Dispatch and Handler interceptors (like discussed [here](../part-iii-infrastructure-components/command-dispatching.md#command-interceptors) for commands, [here](../part-iii-infrastructure-components/event-processing.md#event-interceptors) for events and [here](../part-iii-infrastructure-components/query-processing.md#handler-interceptors) for queries), which intercept a message prior to publishing (Dispatch Interceptor) or whilst it is being handled (Handler Interceptor).
```
Comment (abuijze) [inline part-iv-advanced-tuning/metrics-and-monitoring.md]: It's a reference guide. Let's not make assumptions on what the reader knows ;-)
Suggestion: "There are two types of interceptors [...]"

File: part-iii-infrastructure-components/query-processing.md (hunk 1 above)
Comment (smcvb) [reply part-iii-infrastructure-components/query-processing.md]: Adjusted.

File: part-iv-advanced-tuning/metrics-and-monitoring.md (hunk 2 above)
Comment (smcvb) [reply part-iv-advanced-tuning/metrics-and-monitoring.md]: Daaawh, fair enough, will adjust.

Comment (smcvb) [reply part-iv-advanced-tuning/metrics-and-monitoring.md]: Removed the assumption the reader might be aware of the fact there are two interceptor formats.

File: part-iii-infrastructure-components/event-processing.md [hunk 3]
```diff
@@ -288,3 +238,97 @@ Besides these provided policies, you can define your own. All policies must impl
 
 It is recommended to explicitly define an `ErrorHandler` when using the `AsynchronousEventProcessingStrategy`. The default `ErrorHandler` propagates exceptions, but in an asynchronous execution, there is nothing to propagate to, other than the Executor. This may result in Events not being processed. Instead, it is recommended to use an `ErrorHandler` that reports errors and allows processing to continue. The `ErrorHandler` is configured on the constructor of the `SubscribingEventProcessor`, where the `EventProcessingStrategy` is also provided.
 
+## Event Interceptors
+
+Similarly as with [Command Messages](command-dispatching.md#command-interceptors), Event Messages can also be intercepted prior to publishing and handling to perform additional actions on all Events.
+This thus boils down to same two types of interceptors for messages: the Dispatch- and the HandlerInterceptor.
```
Comment (marleinevankampen) [inline part-iii-infrastructure-components/event-processing.md]: `Handler Interceptor` (space)

**Pull request merged**